src/com/google/javascript/jscomp/parsing/parser/Parser.java

```diff
@@ -2620,16 +2620,29 @@ private boolean peekAssignmentExpression() {
   }
 
   private ParseTree parseAssignment(Expression expressionIn) {
+    return parseAssignment(expressionIn, true);
+  }
+
+  private ParseTree parseAssignment(Expression expressionIn, boolean mayHaveColonType) {
     if (peek(TokenType.YIELD) && inGeneratorContext()) {
       return parseYield(expressionIn);
     }
 
     SourcePosition start = getTreeStartLocation();
-    // TODO(blickly): Allow TypeScript syntax in arrow function parameters
+    GenericTypeListTree generics = maybeParseGenericTypes();
```

Comment:
Isn't this allowing parsing of generics outside arrow functions?

Let's add some tests that generics in random places are forbidden.
e.g.
var x  = <X>(5 + 7);